src/runtime_src/core/common/info_platform.cpp

```diff
@@ -352,6 +352,32 @@ add_clock_info(const xrt_core::device* device, ptree_type& pt)
   }
 }
 
+void
+add_tops_info(const xrt_core::device* device, ptree_type& pt)
+{
+    ptree_type pt_tops_array;
+
+    try {
+        auto resInfo = xrt_core::device_query<xq::xrt_resource_raw>(device);
+        if (resInfo.empty())
+        return;
+
+        size_t count = resInfo.size();
+        for (int i = 0; i < count; i++) {
+            if(resInfo[i].type == xrt_core::query::xrt_resource_raw::resource_type::ipu_tops_max) {
+                ptree_type pt_tops;
+                pt_tops.add("id", xq::xrt_resource_raw::get_name(resInfo[i].type));
+                pt_tops.add("value", resInfo[i].data_double);
+                pt_tops_array.push_back(std::make_pair("", pt_tops));
+            }
+        }
+        pt.put_child("tops", pt_tops_array);
+    }
```

Comment:
Indentation is not following the convention used in the file.
Do not use `camelCase`, instead use `under_score`.
The query return type is a vector, please use range based for-loop

Reply:
camelCase is fixed. 
For now, we only support displaying max TOPs value. In the future, we will expand the supporting list.  We need to loop the result to find out the one value we need.

Reply:
Understood, but please loop using range-based for-loop.
```
for (auto& res : res_info) {
  if (res.type != ...::ipu_tops_max)
    continue;
  ...;
}
```

Reply:
fixed. thanks.